signer/src/transaction_coordinator.rs

```diff
@@ -913,7 +930,9 @@ where
         tokio::pin!(signal_stream);
 
         let future = async {
-            while multi_tx.num_signatures() < wallet.signatures_required() {
+            let mut pending_signers = wallet.public_keys().clone();
+
+            while wallet.public_keys().len() - pending_signers.len() < signatures_required {
```

Comment:
nit: I would add a comment here explaining that this is a "super-condition" wrt `multi_tx.num_signatures() < wallet.signatures_required()`: we remove keys from the pending list (so we only consider the known ones), and we use `recover_ecdsa` on `multi_tx.digest()` so you cannot play as someone else (unless you can sign the correct digest).